main.py

```diff
@@ -648,20 +634,19 @@ def from_network(cls, network: str, seed: bytes, coin: Coin = None, **plan_kwarg
         )
 
     @classmethod
-    def for_demo(cls, original_coin_txid: TxidStr = None) -> 'VaultScenario':
+    def for_demo(cls, original_coin: UtxoStr = None) -> "VaultScenario":
         """
         Instantiate a scenario for the demo, optionally resuming an existing
         vault using the txid of the coin we spent into it.
         """
         coin_in = None
-        if original_coin_txid:
+        if original_coin:
             # We're resuming a vault
-            rpc = BitcoinRPC(net_name="regtest")
-            coin_in = Coin.from_txid(original_coin_txid, 0, rpc)
+            rpc = BitcoinRPC(net_name="signet")
+            txid, outn = original_coin.split(":")
+            coin_in = Coin.from_txid(txid, int(outn), rpc)
 
-        c = VaultScenario.from_network(
-            "regtest", seed=b"demo", coin=coin_in, block_delay=10
-        )
+        c = VaultScenario.from_network("signet", b"demo", coin=coin_in, block_delay=10)
```

Comment:
Hardcoded network constants should be factored out into some kind of `SelectParams`-ish global behavior that still allows use of regtest.

Reply:
`SelectParams` wasn't being used for anything as far as I know, and you already had a global registry of ports for each network.